E0067 needs to be updated to new format

From: src/test/compile-fail/E0067.rs

Error E0067 needs a span_label, updating it from:

```
error[E0067]: invalid left-hand side expression
  --> src/test/compile-fail/E0067.rs:14:5
   |
14 |     LinkedList::new() += 1; //~ ERROR E0368
   |     ^^^^^^^^^^^^^^^^^
```

To:

```
error[E0067]: invalid left-hand side expression
  --> src/test/compile-fail/E0067.rs:14:5
   |
14 |     LinkedList::new() += 1; //~ ERROR E0368
   |     ^^^^^^^^^^^^^^^^^ invalid expression for left-hand side
```

--- a/src/librustc_typeck/check/op.rs
+++ b/src/librustc_typeck/check/op.rs
@@ -41,7 +41,13 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
 
         let tcx = self.tcx;
         if !tcx.expr_is_lval(lhs_expr) {
-            span_err!(tcx.sess, lhs_expr.span, E0067, "invalid left-hand side expression");
+            struct_span_err!(
+                tcx.sess, lhs_expr.span,
+                E0067, "invalid left-hand side expression")
+            .span_label(
+                lhs_expr.span,
+                &format!("invalid expression for left-hand side"))
+            .emit();
         }
     }
 

--- a/src/test/compile-fail/E0067.rs
+++ b/src/test/compile-fail/E0067.rs
@@ -13,4 +13,6 @@ use std::collections::LinkedList;
 fn main() {
     LinkedList::new() += 1; //~ ERROR E0368
                             //~^ ERROR E0067
+                            //~^^ NOTE invalid expression for left-hand side
+                            //~| NOTE cannot use `+=` on type `std::collections::LinkedList<_>`
 }

